urld58fd5e7a9_longrm_weibo_robot_tgz-pJ8-weibo4j_org_json_TestJ8/src/weibo4j/Emotion.java

```diff
@@ -0,0 +1,148 @@
+package weibo4j;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import weibo4j.http.Response;
+import weibo4j.org.json.JSONArray;
+import weibo4j.org.json.JSONException;
+import weibo4j.org.json.JSONObject;
+
+/**
+ * @author SinaWeibo
+ *
+ */
+public class Emotion extends WeiboResponse{
+	private static final long serialVersionUID = -4096813631691846494L;
+
+	private String phrase;
+
+	private String type;
+
+	private String url;
+
+	private boolean is_hot;
+
+	private boolean is_common;
+
+	private int order_number;
+
+	private String category;
+
+	public Emotion(Response res) throws WeiboException {
+		super(res);
+		JSONObject json = res.asJSONObject();
+		try {
+			phrase = json.getString("phrase");
+			type = json.getString("type");
+			url = json.getString("url");
+			is_hot= json.getBoolean("is_hot");
+			order_number = json.getInt("order_number");
+			category = json.getString("category");
+			is_common = json.getBoolean("is_common");
+		} catch (JSONException je) {
+			throw new WeiboException(je.getMessage() + ":" + json.toString(),
+					je);
+		}
+	}
+	public Emotion(JSONObject json) throws WeiboException {
+		try {
+			phrase = json.getString("phrase");
+			type = json.getString("type");
+			url = json.getString("url");
+			is_hot= json.getBoolean("is_hot");
+			order_number = json.getInt("order_number");
+			category = json.getString("category");
+			is_common = json.getBoolean("is_common");
+		} catch (JSONException je) {
+			throw new WeiboException(je.getMessage() + ":" + json.toString(),
+					je);
+		}
+	}
+    static List<Emotion> constructEmotions(Response res) throws WeiboException {
+   	 try {
+            JSONArray list = res.asJSONArray();
+            int size = list.length();
+            List<Emotion> emotions = new ArrayList<Emotion>(size);
+            for (int i = 0; i < size; i++) {
+            	emotions.add(new Emotion(list.getJSONObject(i)));
+            }
+            return emotions;
+        } catch (JSONException jsone) {
+            throw new WeiboException(jsone);
+        } catch (WeiboException te) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An exception is caught and re-thrown within the catch clause without any other operations being performed on the exception, making the catch clause unnecessary.

**Fix**: Perform additional operations on the exception that has been caught. You can wrap a standard exception into a custom exception to add additional information or special handling. When you do that, make sure to set the original exception as the cause. Otherwise, you will lose the stack trace and message of the original exception, which will make it difficult to analyze the event that caused the exception.  
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)